src/triage/component/catwalk/model_trainers.py

```diff
@@ -346,36 +349,48 @@ def process_train_task(
         Returns: (int) model id
         """
-        saved_model_id = retrieve_model_id_from_hash(self.db_engine, model_hash)
-        if (
-            not self.replace
-            and self.model_storage_engine.exists(model_hash)
-            and saved_model_id
-        ):
-            logging.info("Skipping %s/%s", class_path, parameters)
-            return saved_model_id
-
-        if self.replace:
-            reason = "replace flag has been set"
-        elif not self.model_storage_engine.exists(model_hash):
-            reason = "model pickle not found in store"
-        elif not saved_model_id:
-            reason = "model metadata not found"
-
-        logging.info(
-            f"Training {class_path} with parameters {parameters}"
-            f"(reason to train: {reason})"
-        )
         try:
-            model_id = self._train_and_store_model(
-                matrix_store, class_path, parameters, model_hash, misc_db_parameters
-            )
-        except BaselineFeatureNotInMatrix:
-            logging.warning(
-                "Tried to train baseline model without required feature in matrix. Skipping."
+            saved_model_id = retrieve_model_id_from_hash(self.db_engine, model_hash)
+            if (
+                not self.replace
+                and self.model_storage_engine.exists(model_hash)
+                and saved_model_id
+            ):
+                logging.info("Skipping %s/%s", class_path, parameters)
+                if self.run_id:
+                    skipped_model(self.run_id, self.db_engine)
+                return saved_model_id
+
+            if self.replace:
+                reason = "replace flag has been set"
+            elif not self.model_storage_engine.exists(model_hash):
+                reason = "model pickle not found in store"
+            elif not saved_model_id:
+                reason = "model metadata not found"
+
+            logging.info(
+                f"Training {class_path} with parameters {parameters}"
+                f"(reason to train: {reason})"
             )
-            model_id = None
-        return model_id
+            try:
+                model_id = self._train_and_store_model(
+                    matrix_store, class_path, parameters, model_hash, misc_db_parameters
+                )
+            except BaselineFeatureNotInMatrix:
+                logging.warning(
+                    "Tried to train baseline model without required feature in matrix. Skipping."
+                )
+                if self.run_id:
+                    skipped_model(self.run_id, self.db_engine)
+                model_id = None
+            if self.run_id:
+                built_model(self.run_id, self.db_engine)
+            return model_id
+        except Exception as e:
+            errored_model(self.run_id, self.db_engine)
+
+    def flattened_grid_config(self, grid_config):
+        return flatten_grid_config(grid_config)
```

Comment:
ah, because of in `experiments.base`? anyway, can be a `staticmethod`, for now?

Reply:
Yep. Ideally it would be a property, but I figured that moving grid_config to be passed at construction time would be out of scope here. But...I really didn't want to make the experiment have to import more stuff and make this into a responsibility of the component.